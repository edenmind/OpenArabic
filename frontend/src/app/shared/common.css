.icon {
  vertical-align: middle;
}

.center {
  justify-content: center;
  text-align: center;
}

.grid-container {
  margin: 25px;
}

.white-link {
  color: #ffff;
  text-decoration: none;
}

.simple-link {
  text-decoration: none;
}
