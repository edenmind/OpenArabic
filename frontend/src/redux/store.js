import {
  arabicSentenceReducer,
  arabicTextReducer,
  arabicWordsReducer,
  authorReducer,
  categoryReducer,
  englishSentenceReducer,
  englishTextReducer,
  englishWordsReducer,
  sourceReducer,
  textReducer,
  wordByWordReducer,
} from './reducers'

import { configureStore } from '@reduxjs/toolkit'

export default configureStore({
  reducer: {
    title: textReducer,
    category: categoryReducer,
    author: authorReducer,
    source: sourceReducer,
    englishText: englishTextReducer,
    arabicText: arabicTextReducer,
    arabicWords: arabicWordsReducer,
    englishWords: englishWordsReducer,
    arabicSentence: arabicSentenceReducer,
    englishSentence: englishSentenceReducer,
    wordByWord: wordByWordReducer,
  },
})
