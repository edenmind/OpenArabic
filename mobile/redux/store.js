import { configureStore, combineReducers } from '@reduxjs/toolkit'

import { categoriesReducer, textReducer, textsReducer, UIStateReducer, wordsReducer } from './reducers.js'

const rootReducer = combineReducers({
  categories: categoriesReducer,
  text: textReducer,
  texts: textsReducer,
  textLoading: UIStateReducer,
  textsLoading: UIStateReducer,
  arabicFontSize: UIStateReducer,
  englishFontSize: UIStateReducer,
  isTransliterationOn: UIStateReducer,
  arabicFontName: UIStateReducer,
  practicingWords: UIStateReducer,
  words: wordsReducer,
  isDarkMode: UIStateReducer
})

const store = configureStore({
  reducer: rootReducer
})

export { store }
