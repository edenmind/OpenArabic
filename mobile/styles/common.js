import { useMemo } from 'react'
import { StyleSheet } from 'react-native'
import { useSelector } from 'react-redux'
import { paperDarkTheme } from '../constants/paper-theme.js'

const arabicSelector = (state) => state.arabicFontSize
const englishSelector = (state) => state.englishFontSize
const arabicFontNameSelector = (state) => state.arabicFontName

export const useSharedStyles = () => {
  const { arabicFontSize } = useSelector(arabicSelector)
  const { englishFontSize } = useSelector(englishSelector)
  const { arabicFontName } = useSelector(arabicFontNameSelector)

  const englishOpacity = 0.8
  const arabicOpacity = 1
  const paddingBottom = 10
  const horizontalMargin = 33
  const verticalMargin = 33

  const styles = useMemo(() => {
    const arabicFontSizeWeighted = () => {
      if (arabicFontName == 'amiri') {
        return arabicFontSize * 1.2
      }

      if (arabicFontName == 'uthman') {
        return arabicFontSize * 1.5
      }

      if (arabicFontName == 'indopak') {
        return arabicFontSize * 1.75
      }

      if (arabicFontName == 'indopak') {
        return arabicFontSize * 1.5
      }

      if (arabicFontName == 'noto') {
        return arabicFontSize * 1.2
      }
    }

    return StyleSheet.create({
      arabicBody: {
        fontFamily: arabicFontName,
        fontSize: arabicFontSizeWeighted(),
        lineHeight: arabicFontSize * 3.5,
        opacity: arabicOpacity,
        paddingBottom,
        textAlign: 'right',
        writingDirection: 'rtl'
      },
      arabicFooter: {
        fontFamily: 'amiri',
        fontSize: 17,
        opacity: 0.7,
        paddingBottom: 10,
        paddingLeft: 75,
        paddingRight: 75,
        paddingTop: 20,
        readingDirection: 'rtl',
        textAlign: 'center',
        writingDirection: 'rtl'
      },
      button: {
        backgroundColor: paperDarkTheme.colors.onPrimary,
        marginBottom: 10,
        marginTop: 10
      },
      buttonAnswer: {
        backgroundColor: paperDarkTheme.colors.onPrimary,
        marginBottom: 5,
        marginTop: 5
      },
      container: {
        marginBottom: verticalMargin,
        marginLeft: horizontalMargin,
        marginRight: horizontalMargin,
        marginTop: verticalMargin
      },
      divider: {
        marginBottom: 5,
        marginTop: 5
      },
      dividerHidden: {
        marginBottom: 5,
        marginTop: 5,
        opacity: 0
      },
      englishBody: {
        direction: 'ltr',
        fontFamily: 'philosopher',
        fontSize: englishFontSize,
        lineHeight: englishFontSize * 1.5,
        opacity: englishOpacity,
        paddingBottom,
        writingDirection: 'ltr'
      },
      headerContainer: {
        marginLeft: 15,
        marginRight: 15,
        paddingTop: 10
      },
      scrollView: {
        direction: 'rtl',
        marginLeft: horizontalMargin,
        padding: 10,
        writingDirection: 'ltr'
      },
      scrollViewLTR: {
        direction: 'ltr',

        padding: 10,
        writingDirection: 'ltr'
      }
    })
  }, [arabicFontSize, englishFontSize, arabicFontName])

  return styles
}
