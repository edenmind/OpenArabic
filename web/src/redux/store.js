import * as reducers from './reducers'

import { configureStore } from '@reduxjs/toolkit'

export default configureStore({
  reducer: {
    text: reducers.textReducer,
  },
})
