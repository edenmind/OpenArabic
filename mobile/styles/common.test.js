import { expect, describe, it } from '@jest/globals'
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow' // ES6
import { Provider } from 'react-redux'
import { store } from '../redux/store.js'
import { NavigationContainer } from '@react-navigation/native'
import { useSharedStyles } from '../styles/common.js'
import { Text } from 'react-native'

//test that usedSharedStyles exists
describe('useSharedStyles', () => {
  it('exists', () => {
    //provide a mock store
    const renderer = new ShallowRenderer()
    renderer.render(
      <Provider store={store}>
        <NavigationContainer>
          <useSharedStyles />
        </NavigationContainer>
      </Provider>
    )

    expect(useSharedStyles).toBeDefined()
  })

  it('returns an object', () => {
    //provide a mock store
    const renderer = new ShallowRenderer()
    renderer.render(
      <Provider store={store}>
        <NavigationContainer>
          <useSharedStyles />
        </NavigationContainer>
      </Provider>
    )

    expect(typeof useSharedStyles).toBe('function')
  })

  it('should match snapshot', () => {
    //create a basic component to test
    const TestComponent = () => {
      const sharedStyle = useSharedStyles()
      return (
        <div>
          <Text style={sharedStyle.arabicBody}>Test</Text>
        </div>
      )
    }

    //provide a mock store
    const renderer = new ShallowRenderer()

    renderer.render(
      <Provider store={store}>
        <NavigationContainer>
          <TestComponent />
        </NavigationContainer>
      </Provider>
    )

    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
