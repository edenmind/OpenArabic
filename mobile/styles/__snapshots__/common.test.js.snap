// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useSharedStyles should match snapshot 1`] = `
<Context.Provider
  value={
    {
      "getServerState": undefined,
      "noopCheck": "once",
      "stabilityCheck": "once",
      "store": {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ForwardRef(NavigationContainerInner)>
    <TestComponent />
  </ForwardRef(NavigationContainerInner)>
</Context.Provider>
`;
